src/ocean/math/BinaryHistogram.d

```diff
@@ -0,0 +1,393 @@
+/*******************************************************************************
+
+    An integer, binary logarithmic statistics counter
+
+    Collects the following statistical information:
+    - a binary logarithmic histogram with bins from ≥1 to <1<<MaxPow2
+      (see template documentation, below), one bin per power of two, plus one
+      bin for each 0 and ≥1<<MaxPow2,
```

Comment:
A small thing: Replace `1<<x` with `2^x`, this is about powers of 2.

Reply:
Updated.